src/System.Linq/src/System/Linq/Contains.cs

```diff
@@ -12,16 +11,34 @@ public static partial class Enumerable
         public static bool Contains<TSource>(this IEnumerable<TSource> source, TSource value)
         {
             ICollection<TSource> collection = source as ICollection<TSource>;
-            if (collection != null) return collection.Contains(value);
+            if (collection != null)
+            {
+                return collection.Contains(value);
```

Comment:
This is how it's been for a long time, and it yields better performance, so I don't think we should change it.  Interestingly, MSDN is a bit inconsistent on the subject.  On the one hand, the docs explicitly say that if the enumerable is an ```ICollection<T>``` that its ```Contains``` will be used... on the other hand, it states that the default compare is used.